libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -71,10 +83,57 @@ func (p *addHostMetadata) loadData() {
 		if p.info.OS.Build != "" {
 			p.data.Put("host.os.build", p.info.OS.Build)
 		}
+
+		if p.config.NetInfoEnabled {
+			// IP-address and MAC-address
+			var ipList, hwList = p.getNetInfo()
+			p.data.Put("host.ip", ipList)
+			p.data.Put("host.mac", hwList)
+		}
+
 		p.lastUpdate = time.Now()
 	}
 }
 
+func (p addHostMetadata) getNetInfo() ([]string, []string) {
+	var ipList []string
+	var hwList []string
+
+	// Get all interfaces and loop through them
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return ipList, hwList
```

Comment:
Instead or returning the empty lists could we return here the error? Meaning it would return `nil, nil, err`. On line 89 we could then check for the error and would not add the fields in case of an error and could also log the error.

Reply:
Sure, do you have a code example of logging, that I could look at?